shared/teams/edit-team-welcome-message/index.tsx

```diff
@@ -0,0 +1,125 @@
+
+const EditTeamWelcomeMessage = (props: Props) => {
+  const teamID = Container.getRouteProps(props, 'teamID', Types.noTeamID)
+
+  if (teamID === Types.noTeamID) {
+    throw new Error(`There was a problem loading the welcome message page, please report this error.`)
+  }
+
+  const waitingKey = Container.useSelector(_ => Constants.setWelcomeMessageWaitingKey(teamID))
+  const waiting = Container.useAnyWaiting(waitingKey)
+  const error = Container.useSelector(state => state.teams.errorInEditWelcomeMessage)
+  const origWelcomeMessage = Container.useSelector(state => {
+    const message = Constants.getTeamWelcomeMessageByID(state, teamID)
+    if (message) {
+      return message
+    }
+    throw new Error(`There was a problem loading the welcome message, please report this error.`)
+  })
+
+  const [welcomeMessage, setWelcomeMessage] = React.useState(origWelcomeMessage)
+
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const onSave = () => dispatch(TeamsGen.createSetWelcomeMessage({message: welcomeMessage, teamID}))
+  const onClose = () => dispatch(nav.safeNavigateUpPayload())
+
+  const wasWaiting = Container.usePrevious(waiting)
+  React.useEffect(() => {
+    if (!waiting && wasWaiting && !error) dispatch(nav.safeNavigateUpPayload())
+  }, [waiting, wasWaiting, nav, dispatch, error])
+  return (
+    <Kb.Modal
+      mode="Default"
+      banners={
+        error
+          ? [
+              <Kb.Banner color="red" key="err">
+                {error}
+              </Kb.Banner>,
+            ]
+          : undefined
+      }
+      onClose={onClose}
+      footer={{
+        content: (
+          <Kb.ButtonBar fullWidth={true} style={styles.buttonBar}>
+            <Kb.Button style={styles.button} label="Cancel" onClick={onClose} type="Dim" />
+            <Kb.Button
+              style={styles.button}
+              disabled={
+                welcomeMessage.text === origWelcomeMessage.text &&
+                welcomeMessage.set === origWelcomeMessage.set
+              }
+              label="Save"
+              onClick={onSave}
+              waiting={waiting}
+            />
+          </Kb.ButtonBar>
+        ),
+      }}
+      header={{
+        title: 'Edit welcome note',
+      }}
+    >
+      <Kb.Box2 alignItems="flex-start" direction="vertical" style={styles.container}>
+        <Kb.LabeledInput
+          placeholder="Welcome note"
+          onChangeText={x => setWelcomeMessage({set: true, text: x})}
+          value={computeWelcomeMessageText(welcomeMessage, false /* cannotWrite */)}
+          multiline={true}
+          rowsMin={3}
+          rowsMax={3}
+          maxLength={welcomeMessageMaxLen}
+          autoFocus={true}
+        />
+        <Kb.Text type="BodySmall" style={styles.info}>
+          {welcomeMessage.set && welcomeMessage.text.length === 0 ? (
+            'No welcome note will be shown to new members.'
+          ) : (
+            <>&nbsp;</>
```

Comment:
why this rather than not showing the `Kb.Text` if the welcome message is set?

Reply:
So the space under the text box doesn't change when the message doesn't appear.

Reply:
okay, in that case I think I'd do `? (blah) : (' ')` - if you wrap it in a fragment (`<>` / `</>`) like that you might have issues on mobile